sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/consumer/strategies/retryprocessingerror/RetryProcessingErrorMLS.java

```diff
@@ -27,12 +28,27 @@ public class RetryProcessingErrorMLS<K, V> extends MessageListenerStrategy<K, V>
 
   private static final Logger LOGGER = LoggerFactory.getLogger(RetryProcessingErrorMLS.class);
   private final MessageHandler<K, V> handler;
-  private final ErrorHandler<K, V> errorHandler;
+  private final ErrorHandler<K, V> intermediateErrorHandler;
+  private final ErrorHandler<K, V> finalErrorHandler;
+  private final long maxRetryCount;
   private String consumerName;
+  private Map<TopicPartition, OffsetCounter> offsetCounters =
```

Comment:
Does `TopicPartition` have an appropriate equals method, so that the partition is matched reliable? To be safe, we could just go with the number.

NIT: I prefer more precise names for such maps, e.g. `offsetCountersByPartition`.

Reply:
Yes, it has